games/hangman.c

```diff
@@ -0,0 +1,277 @@
+
+            for (i = 0; i < (strlen(current_word) - 1); i++) {
+                printf("%c ", hidden[i]);
+            }
+
+            if ((strlen(guesses)) > 0) {
+                printf("\nSo far, you have guessed: ");
+                for (i = 0; i < (strlen(guesses)); i++) {
+                    printf("%c ", guesses[i]);
+                }
+            }
+
+            printf("\nYou have %d guesses left.", (13 - incorrect));
+            printf("\nPlease enter a letter: ");
+            scanf(" %c", &guess);
+
+            guesses_size = sizeof(guesses) / sizeof(char);
+
+        } while (new_guess(guess, guesses, guesses_size) != -1);
+
+        guesses[strlen(guesses)] = guess; // adds new letter to guesses array
+
+        if (in_word(guess, current_word, strlen(current_word)) == 1) {
+            printf("That letter is in the word!");
+            for (i = 0; i < strlen(current_word); i++) {
+                if (tolower(current_word[i]) == guess) {
+                    hidden[i] = guess;
+                }
+            }
+        } else {
+            printf("That letter is not in the word.\n");
+            incorrect++;
+        }
+        picture(incorrect);
+    }
+
+    won(incorrect);
+    return 0;
+}
+
+/**
+ * @brief checks if letter has been guessed before
+ * @param new_guess letter that has been guessed by player
+ * @param guesses array of player's previous guesses
+ * @param size size of guesses[] array
+ * @returns 1 if letter has been guessed before
+ * @returns -1 if letter has not been guessed before
+ */
+int new_guess(char new_guess, const char guesses[], int size) {
+
+    int j;
+
+    for (j = 0; j < size; j++) {
+        if (guesses[j] == new_guess) {
+            printf("\nYou have already guessed that letter.");
+            return 1;
+        }
+    }
+    return -1;
+}
+
+/**
+ * @brief checks if letter is in current word
+ * @param letter letter guessed by user
+ * @param word current word
+ * @param size length of word
+ * @returns 1 if letter is in word
+ * @returns -1 if letter is not in word
+ */
+int in_word(char letter, char word[], unsigned int size) {
+
+    int i;
+
+    for (i = 0; i < size; i++) {
+        if (tolower(word[i]) == letter) {
+            return 1;
+        }
+    }
+
+    return -1;
+}
+
+/**
+ * @brief creates a new game - generates a random word and stores in global variable current_word
+ * @returns void
+ */
+void new_game() {
+
+    int line_number = 0;
+    int random_num;
+    int s = 0;
+
+    FILE *fptr;
+    fptr = fopen("words.txt", "r");
+
+    // counts number of words in file - assumes each word on new line
+    while (fgets(current_word, 100, fptr) != NULL) {
+        line_number++;
+    }
+    fclose(fptr);
+
+    srand(time(NULL));
+    random_num = rand() % line_number;
+
+    // selects randomly generated word
+    fptr = fopen("words.txt", "r");
+    while (s < random_num){
+        fgets(current_word, 100, fptr);
+        s++;
+    }
+    fclose(fptr);
```

Comment:
```suggestion
    current_word[strlen(current_word) - 1] = '\0';
    fclose(fptr);
```

Reply:
...In case that you haven't noticed yet, there's no need to close this one then open a new PR, just push your fixes to your fork then they will appear in this PR.